rudder-core/src/main/resources/reportsSchema.sql

```diff
@@ -456,6 +456,18 @@ start_minute int,
 startTime timestamp with time zone default now(),
 endTime timestamp with time zone
 );
+
+-- Create the table for the node configuration
+CREATE TABLE dataSources (
+  id            text NOT NULL CHECK (id <> '')  
+
+-- data source properties and status are valid json, until we can use postgres 9.2 keep text type (more details in configuration details)
+
+, properties     text 
+, status     text 
```

Comment:
I believe they must not be empty, else it can't be changed to json latter on. So, nullable, but with a constrain for empty string